### PR TITLE
:sparkles: Added token whitelisting for treasuries

### PR DESCRIPTION
**Pull request by acemasterjb**
## Summary
Addresses issue #27.

## Details
Some tokens a treasury may hold (most likely airdropped) can potentially corrupt the statistics that Whip generates. These tokens are typically spam tokens, which (from testing various treasuries) either causes issues for the historical token pricefeed for these tokens obtained from Coingecko or historical treasury balances obtained from Covalent.

Since these are spam tokens, which are likely not going to be traded on reputable exchanges, it's safe to assume that they will not be listed on [TokenLists](https://tokenlists.org). This PR utilises TokenList to filter spam tokens through a whitelist, thus mitigating the issues that #27 raises.

### CODE REVIEW

**Comment by acemasterjb** [comment]
So far this PR utilises only the CoinGecko token list. I'm going to add CoinMarketCap and monitor the results of the addition.

@lajarre @vaughnmck any more token lists you think we should add?

**Comment by vaughnmck** [comment]
I vote for those two for now

File: backend/app/libs/storage_helpers.py [hunk 1]
```diff
@@ -19,6 +19,17 @@ def retrieve_treasuries_metadata() -> list[tuple[str, int]]:
     return [tuple(json.loads(t).values()) for t in db.lrange("treasuries", 0, -1)]
 
 
+def store_token_whitelist(address: str):
```
Comment (lajarre) [inline backend/app/libs/storage_helpers.py:22]: Can't we rather use a Redis set?

👓 https://redis.io/docs/manual/data-types/

File: backend/app/__init__.py [hunk 2]
```diff
@@ -4,11 +4,11 @@
 import sentry_sdk
 
 if "REDIS_TLS_URL" in os.environ:
-    db = redis.Redis.from_url(os.environ["REDIS_TLS_URL"], ssl_cert_reqs=None)
+    db = redis.Redis.from_url(os.environ["REDIS_TLS_URL"], ssl_cert_reqs=None, decode_responses=True)
```
Comment (lajarre) [inline backend/app/__init__.py:7]: This seems good for cases where we're storing strings, but what about when we're storing JSONs? I guess `json.loads` both works with bytes and strings right?

File: backend/app/libs/tasks/get_assets.py [hunk 3]
```diff
@@ -34,8 +36,13 @@
 
 
 async def make_treasury(treasury_address: str, chain_id: int) -> Treasury:
+    token_whitelist = retrieve_token_whitelist()
```
Comment (lajarre) [inline backend/app/libs/tasks/get_assets.py:39]: Nitpicking: I'm ok with this, but let's try keeping `get_assets` module as simple as possible and not concerned with implem details. I would rather put these 3 lines in a separate function in the `tokenlists` module.

**Review by lajarre — CHANGES_REQUESTED** [review]

File: backend/app/libs/storage_helpers.py (hunk 1 above)
Comment (acemasterjb) [reply backend/app/libs/storage_helpers.py:22]: Thanks, this will give us a big performance improvements 👍.

File: backend/app/__init__.py (hunk 2 above)
Comment (acemasterjb) [reply backend/app/__init__.py]: Actually, as far as I know `json.loads` only works with strings. I'm surprised we haven't had to decode bytes responses from Redis till now.

**Pull request closed without merging**